src/component.vue

```diff
@@ -90,18 +83,24 @@ export default {
     }
   },
   watch: {
-    value(newValue) {
-      // avoid trigering the directive's update hook when we emit
-      // the unmasked value to the parent component
-      if (newValue !== this.emittedValue) {
-        this.maskedValue = newValue
+    modelValue: {
+      deep: true,
```

Comment:
do we need deep on the modelValue?  Since this is an input the value should be a primitive type.

Reply:
yes, thank you good catch. the watch breaking change is on arrays so will remove it here

- [ ] remove deep for modelValue